packages/api/src/controllers/asset.ts

```diff
@@ -1089,11 +1110,64 @@ app.delete("/:id", authorizer({}), async (req, res) => {
   if (!req.user.admin && req.user.id !== asset.userId) {
     throw new ForbiddenError(`users may only delete their own assets`);
   }
+
+  if (asset.status.phase === "deleting" || asset.deleted) {
+    throw new BadRequestError(`Asset is already deleted`);
+  }
+
   await req.taskScheduler.deleteAsset(asset);
   res.status(204);
   res.end();
 });
 
+app.put("/:id/restore", authorizer({}), async (req, res) => {
+  const { id } = req.params;
+  const asset = await db.asset.get(id);
+
+  if (!asset) {
+    throw new NotFoundError(`Asset not found`);
```

Comment:
I believe our errors are generally `lower case`, let's keep it consistent (other occurrences in this PR)